src/irmin-pack/unix/gc_worker.ml

```diff
@@ -122,7 +118,24 @@ module Make (Args : Gc_args.S) = struct
     stats := Gc_stats.Worker.add_file_size !stats "old_prefix" prefix_size;
     stats := Gc_stats.Worker.add_file_size !stats "old_mapping" mapping_size
 
-  let run ~generation root commit_key new_prefix_end_offset =
+  type new_suffix = {
+    start_offset : int63;
+    chunk_start_idx : int;
+    chunk_num : int;
+    dead_bytes : int63;
+  }
+  [@@deriving irmin]
+
+  type gc_results = {
+    suffix : new_suffix;
+    chunk_idx_gced : int list;
+    stats : Stats.Latest_gc.worker;
+  }
+  [@@deriving irmin]
+
+  type gc_output = (gc_results, Args.Errs.t) result [@@deriving irmin]
+
+  let run ~generation root commit_key new_suffix_start_offset =
```

Comment:
This function used  `new_prefix_end_offset` instead of `new_suffix_start_offset`. The two used to coincide I think. Is it not the case anymore? (I'm wondering if this change is refactoring or something deeper that I missed).

Reply:
Yeah, they are the same value. I changed to this name because it is only used for the suffix part and felt better to reference a suffix value than a prefix value (and in my mind GC is an operation that primarily moves the suffix's starting offset). It's nothing deeper than trying to clarify naming as I go. :smile:

Reply:
ok, how about renaming it into `suffix_start_offset` then? since there is no `new_suffix` and it is the `suffix_start_offset`  from the gc.ml code

Reply:
I think I share the perspective of @Ngoguey42. I read this variable as `new_"suffix_start_offset"` (ie the value after GC) rather than `"new_suffix"_start_offset`, but either work in indicating it is not the current value :sweat_smile:.

Related to this naming scheme (and your mention of `gc.ml` @icristescu), I removed the `new_` prefix from file manager's swap (see https://github.com/mirage/irmin/pull/2126/files#diff-7c232e5fd0a8196916e9a74fba7bddf6c15b172369747d9883eb82a94eb2ef4fR244-R247) because 1) the generation didn't have it and 2) it is called after GC so the `new_` is implicit when setting these values (and why I suppose `generation` did not use it already).

A summary of why I think the current naming works:
1)The  `gc_worker.ml` code has not yet completed the GC, so it uses `new_`. It is looking to the future.
2) The `gc.ml` code in question is after GC completed (finalisation), so it does not. It is setting the present. (one place I could see using `new_` in `gc.ml` code is what we currently call `t.offset` -- this could be possibly renamed to `new_suffix_start_offset`).

Reply:
I renamed `t.offset` to `new_suffix_start_offset` in `gc.ml` to at least be consistent. This feels good to me for now. [`ad270b8` (#2126)](https://github.com/mirage/irmin/pull/2126/commits/ad270b846b65bd57ab670a8703c7c470a8d1ee1d)

Reply:
thanks